### ISSUE TITLE
Make the ArrayFire testsuite a separate project

### ISSUE THREAD

**Issue by ghisvail** [arrayfire/arrayfire#1108]
As for the [examples](https://github.com/arrayfire/arrayfire/blob/devel/examples/CMakeLists.txt), it would be really nice to make the testsuite a separate project, and to support building the tests either with the ArrayFire targets or existing binaries.

The specific use case I have in mind concerns continuous integration inside Linux distributions [1][2], whereby binary packages gets automatically tested on changes to their corresponding dependencies.

[1] https://ci.debian.net
[2] https://fedoraproject.org/wiki/Koschei

**Comment by pavanky** [comment]
@ghisvail From what I understand, you want the ability to build and test the test files from a fresh checkout (of arrayfire) without actually building the libraries (and perhaps use libraries present in the system already).

Is this a fair assessment?

**Comment by ghisvail** [comment]
> the ability build and test the test files from a fresh checkout (of arrayfire)

> and perhaps use libraries present in the system already

Exactly.

Ideally, from an arbitrary location with write access, I'd like to be able to do:
```
cmake $(ArrayFire_sourcedir)/test
make
make test  # or something more specific with ctest
```
and get CMake to use the packaged version of ArrayFire (via `find_package`).

**Comment by pavanky** [comment]
Fixed in hotfixes-3.2.1
